RTNeural/activation/activation_eigen.h

```diff
@@ -464,6 +464,418 @@ class PReLUActivationT
+        : Activation<T>(size, {}, "swish"), beta(beta)
+    {
+        inVec = Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic>::Zero(size, 1);
+        outVec = Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic>::Zero(size, 1);
+    }
+
+    SwishActivation(std::initializer_list<int> sizes, T beta = (T)1)
+        : SwishActivation(*sizes.begin(), beta)
+    {
+    }
+
+    RTNEURAL_REALTIME inline void forward(const T* input, T* out) noexcept override
+    {
+        inVec = Eigen::Map<const Eigen::Matrix<T, Eigen::Dynamic, 1>, RTNeuralEigenAlignment>(
+            input, Layer<T>::in_size, 1);
+
+        /** apply swish: x * sigmoid(beta * x) */
+        outVec = inVec.array() * MathsProvider::sigmoid(beta * inVec.array());
+        std::copy(outVec.data(), outVec.data() + Layer<T>::in_size, out);
+    }
+
+    Eigen::Matrix<T, Eigen::Dynamic, 1> inVec;
+    Eigen::Matrix<T, Eigen::Dynamic, 1> outVec;
+
+private:
+    T beta;
+};
+
+/** Static implementation of a Swish activation layer. */
+template <typename T, int size, typename MathsProvider = DefaultMathsProvider>
+class SwishActivationT
+{
+    using v_type = Eigen::Matrix<T, size, 1>;
+
+public:
+    static constexpr auto in_size = size;
+    static constexpr auto out_size = size;
+
+    /** set optional beta parameter (default is 1) */ 
+    SwishActivationT(T beta = (T)1)
+        : outs(outs_internal), beta(beta)
+    {
+        outs = v_type::Zero(); // initialize output vector to zero
+    }
+
+    std::string getName() const noexcept { return "swish"; }
+
+    constexpr bool isActivation() const noexcept { return true; }
+
+    RTNEURAL_REALTIME void reset() { }
+
+    RTNEURAL_REALTIME inline void forward(const v_type& ins) noexcept
+    {
+        /** apply swish: x * sigmoid(beta * x) */
+        outs = ins.array() * MathsProvider::sigmoid(beta * ins.array());
+    }
+
+    Eigen::Map<v_type, RTNeuralEigenAlignment> outs;
+
+private:
+    T outs_internal alignas(RTNEURAL_DEFAULT_ALIGNMENT)[out_size];
+    T beta;
+};
+
+/** Dynamic implementation of a Softplus activation layer. */
+template <typename T, typename MathsProvider = DefaultMathsProvider>
+class SoftplusActivation : public Activation<T>
+{
+public:
+    explicit SoftplusActivation(int size)
+        : Activation<T>(size, {}, "softplus")
+    {
+        inVec = Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic>::Zero(size, 1);
+        outVec = Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic>::Zero(size, 1);
+    }
+
+    SoftplusActivation(std::initializer_list<int> sizes)
+        : SoftplusActivation(*sizes.begin())
+    {
+    }
+
+    RTNEURAL_REALTIME inline void forward(const T* input, T* out) noexcept override
+    {
+        inVec = Eigen::Map<const Eigen::Matrix<T, Eigen::Dynamic, 1>, RTNeuralEigenAlignment>(
+            input, Layer<T>::in_size, 1);
+
+        /** Apply Softplus: ln(1 + exp(x)) */
+        outVec = (1 + MathsProvider::exp(inVec.array())).log(); //log() awaiting mathsprovider
```

Comment:
Yeah, we should all `log` to the `MathsProvider` if we're using it here. The pattern implemented [here](https://github.com/jatinchowdhury18/RTNeural/blob/main/RTNeural/maths/maths_eigen.h) should be pretty straightforward, but I'm happy to assist with that if you need.